ydb/docs/en/core/concepts/glossary.md

```diff
@@ -0,0 +1,515 @@
+
+### Topic {#topic}
+
+A **topic** is a persistent queue that can be used for reliable asynchronous communications between various systems via message passing. {{ ydb-short-name }} provides the infrastructure to ensure "exactly once" semantics in such communications, which ensures that there are both no lost messages and no accidental duplicates.
+
+Several terms related to topics are listed below. How {{ ydb-short-name }} topics work is explained in more detail in a separate article [{#T}](topic.md).
+
+#### Partition {#partition}
+
+For horizontal scaling purposes, topics are divided into separate elements called **partitions**. Thus, a partition is a unit of parallelism within a topic. Messages inside each partition are ordered.
+
+However, subsets of data managed by a single [data shard](#data-shard) or [column shards](#column-shard) can also be called partitions.
+
+#### Offset {#offset}
+
+An **offset** is a sequence number that identifies a message inside a [partition](#partition).
+
+#### Producer {#producer}
+
+A **producer** is an entity that writes new messages to a topic.
+
+#### Consumer {#consumer}
+
+A **consumer** is an entity that reads messages from a topic.
+
+### Change data capture {#cdc}
+
+**Change data capture** or **CDC** is a mechanism that allows subscribing to a stream of changes to a given [table](#table). Technically, it is implemented on top of [topics](#topic). It is described in more detail in a separate article [{#T}](cdc.md).
+
+### YQL {#yql}
+
+**YQL ({{ ydb-short-name }} Query Language)** is a high-level language for working with the system. It is a dialect of [ANSI SQL](https://en.wikipedia.org/wiki/SQL). There's a lot of content covering YQL, including a [tutorial](../dev/yql-tutorial/index.md), [reference](../yql/reference/syntax/index.md), and [recipes](../recipes/yql/index.md).
+
+### Federated queries {#federated-queries}
+
+**Federated queries** is a feature that allows querying data stored in systems external to the {{ ydb-short-name }} cluster.
+
+A few terms related to federated queries are listed below. How {{ ydb-short-name }} federated queries work is explained in more detail in a separate article [{#T}](federated_query/index.md).
+
+#### External data source {#external-data-source}
+
+An **external data source** or **external connection** is a piece of metadata that describes how to connect to a supported external system for federated query execution.
+
+#### External table {#external-table}
+
+An **external table** is a piece of metadata that describes a particular dataset that can be retrieved from an [external data source](#external-data-source).
+
+#### Secret {#secret}
+
+A **secret** is a sensitive piece of metadata that requires special handling. Typically, secrets are used in [external data source](#external-data-source) definitions and represent things like passwords and tokens.
```

Comment:
Also secrets can be used for
* async replication
* data offloading to S3

Reply:
```suggestion
A **secret** is a sensitive piece of metadata that requires special handling. For example, secrets can be used in [external data source](#external-data-source) definitions and represent things like passwords and tokens.
```
As far as I know, both of these features are not documented yet. So for now, I've just changed: typically -> for example.